src/SessionMessages/SessionMessage/MessageFiles.tsx

```diff
@@ -12,23 +12,63 @@ interface MessageFilesProps extends PropsWithChildren {
   files: ConversationFile[];
 }
 
+
 export const MessageFiles: FC<MessageFilesProps> = ({ files, children }) => {
   const { theme } = useContext(ChatContext);
   const Comp = children ? Slot : MessageFile;
+  const [expanded, setExpanded] = useState<boolean>(false);
 
   if (!files || files.length === 0) {
     return null;
   }
 
+  // Filter image files
+  const imageFiles = files.filter(file => file.type.startsWith('image/'));
+  const otherFiles = files.filter(file => !file.type.startsWith('image/'));
+
   return (
-    files.length > 0 && (
-      <div className={cn(theme.messages.message.files.base)}>
-        {files.map((file, index) => (
-          <Comp key={index} {...file}>
-            {children}
-          </Comp>
-        ))}
-      </div>
-    )
+    <>
+      {imageFiles.length > 3 ? (
```

Comment:
Instead of these nested ternaries, I think we can simplify here by doing something like 
```
{ imageFiles.length <=3 || expanded ?
    imageFiles.map(file => <Comp />) :
    imageFiles.slice(0,3).map(file => <figure />)
} 
```

we probably also want to consider making the <figure /> another slot so that this functionality is also customizable